### ISSUE TITLE
Decide on how to render tens of thousands of locations on map

### ISSUE THREAD

**Issue by boulderalf** [BeeSafeEarth/BeeSafeEarth#4]
<a href="https://github.com/boulderalf"><img src="https://avatars.githubusercontent.com/u/2962724?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [boulderalf](https://github.com/boulderalf)**
_Monday Mar 21, 2016 at 22:31 GMT_
_Originally opened as https://github.com/boulderalf/BeeSafeEarth/issues/4_

---

We will hopefully have tens of thousands of pledged locations in our system.  How do we show those on the map when a user is zoomed out to view an entire continent?  Having 10's of thousands of points in the Google Maps API can have serious performance implications.

Let's use this issue to track ideas for scalable rendering of lots of points.

cc @ZebGirouard

**Comment by boulderalf** [comment]
<a href="https://github.com/boulderalf"><img src="https://avatars.githubusercontent.com/u/2962724?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [boulderalf](https://github.com/boulderalf)**
_Monday Mar 21, 2016 at 22:32 GMT_

---

https://developers.google.com/maps/articles/toomanymarkers#introduction

I think we will need some kind of server-based clustering algorithm.  Any other ideas?

**Comment by boulderalf** [comment]
<a href="https://github.com/boulderalf"><img src="https://avatars.githubusercontent.com/u/2962724?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [boulderalf](https://github.com/boulderalf)**
_Friday Mar 25, 2016 at 13:09 GMT_

---

@ZebGirouard it looks like your [Hello World example with Fusion Tables](https://github.com/boulderalf/BeeSafeEarth/issues/1#issuecomment-201112655) can easily solve our "render thousands of icons on one map" challenge.

cc @DaveLaskarzewski @autumnmartin

**Comment by boulderalf** [comment]
<a href="https://github.com/ZebGirouard"><img src="https://avatars.githubusercontent.com/u/5480464?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ZebGirouard](https://github.com/ZebGirouard)**
_Friday Mar 25, 2016 at 14:23 GMT_

---

@boulderalf , my second [KML Hello World example](http://zebgirouard.github.io/GoogleMapsAPI_Sandbox/beeMapCustom.html) has a hidden "benefit" of having its KML file (aka the points with their styles) cached on Google's servers.  So we wouldn't be doing any of the heavy lifting on map load.

A seemingly solid plan would be to have our pledge captains in various locations have their own sheets (or SQL tables, etc).  Then we script a sanitization (just lat and long) and conversion to KML for ALL locations as well as a high-level KML with just the geographical center of all "captained locations" (Boulder, Boston, Portland, etc).  We could then switch the KML layer depending on our zoom level with an event listener.  The cool thing about KML is that it only loads the points in view...[see this link here](https://productforums.google.com/forum/#!topic/earth/07q-jofuPo0).

**Comment by boulderalf** [comment]
@ZebGirouard here's another example of clustering large point sets using MapBox (https://www.mapbox.com/blog/supercluster/?utm_source=newsletter_april&utm_medium=email&utm_content=supercluster&utm_campaign=newsletter_april)

**Comment by ZebGirouard** [comment]
@boulderalf , yeah, it seems nice. At a certain point soon, we should probably decide if we're going down the MapBox or the GoogleMaps route.  Investigating two options the whole time is going to take us twice the time.

**Comment by DaveLaskarzewski** [comment]
Pros / Cons list?

**Comment by Beesafedavid** [comment]
@boulderalf @ZebGirouard I agree that a decision should be made. I'm happy to do a hangout to discuss, if that is the best approach.
As per Zeb's and my meeting yesterday, we hope to have the following:
1. Individual pledges
2. BSNs, in-progress (with name, pledges to date, etc. in legend, and greened background) and Certified (with beehive representing on map, and legend still available on roll-over. Pin-drop with a radius for BSNs seems like it might be easiest if we want to remain automatic.
3. BS Regions/Cities available on roll-over. 
4. BS Businesses (Retailers, Landscapers, etc.) layer available at Region/City level.
5. other?
